backend/src/notes/notes.service.ts

```diff
@@ -102,9 +103,10 @@ export class NotesService {
 
     // We cast to a note early to keep the later code clean
     const newNote = Note.create(owner, alias) as Note;
+
+    const patch = createPatch(newNote.publicId, '', noteContent);
     newNote.revisions = Promise.resolve([
-      //TODO: Calculate patch
-      Revision.create(noteContent, noteContent, newNote) as Revision,
+      Revision.create(noteContent, patch, newNote) as Revision,
```

Comment:
shouldn't we call the `revisionsService` here?

Reply:
I've been thinking about that too. But there are some other problems.
- The revision service method saves the creates revision entity, which is not possible at this point because the note doesnt exist.
- getLatestRevision throws an error if the provided note doesn't have an id or any revisions. (The whole "lets cast the ENTITY_WITHOUT_ID to ENTITY" behaviour is firing back now :sweat_smile: )

I would prefer to keep it this way and do a major refactoring of the whole `.create() as ENTITY` code.